lib/base-controller.js

```diff
@@ -23,6 +23,41 @@ function getErrorLength() {
   }
 }
 
+/*
+ * Utility function which returns the
+ * name of the step a field is present on
+ */
+function getStepFromFieldName(fieldName, steps) {
+  return _.findKey(steps, function findKey(step) {
+    if (!step.fields) {
+      return false;
+    }
+    return step.fields.indexOf(fieldName) > -1;
+  });
+}
+
+/*
+ * Utility function to create an array
+ * of field objects made up of field name,
+ * value and step
+ */
+function getFieldObjectsFromNames(fields, values, modifiers, steps) {
+  return fields.filter(function filter(item) {
+    return (values[item] !== undefined && values[item] !== '') || modifiers[item] !== undefined;
```

Comment:
not in any of the examples I tried, ideally it should always be `undefined` but it is set to an empty string in the /how step with the custom radio - this will probably be fixed if my PR to template mixins gets merged - https://github.com/UKHomeOffice/passports-template-mixins/pull/57